cmd/kured/main.go

```diff
@@ -788,6 +788,18 @@ func rebootAsRequired(nodeID string, booter reboot.Reboot, sentinelCommand []str
 		if rebootDelay > 0 {
 			log.Infof("Delaying reboot for %v", rebootDelay)
 			time.Sleep(rebootDelay)
+
+			log.Infof("Making sure %s is drained before reboot", node.GetName())
+			err = drain(client, node)
```

Comment:
I don' think we should trigger another drain. 

The state should be as _expected_ before reboot, but not triggering an extra drain. See @jackfrancis 's opinion on the issue.